sdk/identity/azure-identity/inc/azure/identity/default_azure_credential.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Default Azure Credential.
+ */
+
+#pragma once
+
+#include <azure/core/credentials/token_credential_options.hpp>
+
+#include <azure/identity/chained_token_credential.hpp>
+
+#include <memory>
+
+namespace Azure { namespace Identity {
+
+  /**
+   * @brief Default Azure Credential combines multiple credentials that depend on the setup
+   * environment and require no parameters into a single chain. If the environment is set up
+   * sufficiently for at least one of such credentials to work, `DefaultAzureCredential` will work
+   * as well.
+   *
+   * @details This credential is using the #ChainedTokenCredential of 3 credentials in the order:
+   * #EnvironmentCredential, #AzureCliCredential, and #ManagedIdentityCredential. Even though the
+   * credentials being used and their order is documented, it may be changed in the future versions
+   * of the SDK, potentially bringing breaking changes in its behavior.
+   *
+   * @note This credential is intended to be used at the early stages of development, to allow the
+   * developer some time to work with the other aspects of the SDK, and later to replace this
+   * credential with the exact credential that is the best fit for the application. It is not
+   * intended to be used in a production environment.
```

Comment:
really!!  This is the first time I heard about this recommendation.!!  Is this documented within some Identity docs?

Reply:
Yes - this PR has updates to README.md also mentioning it.
We've had a meeting on designing this credential, and that is the message that we wanted to use. I can share the recording with you.
C++ SDK will be better in this way - other SDKs released this credential without this disclaimer, and can't change it now. We build on their experience. If they were adding it today, they probably would've went with the same user contract.